apps/hashdotdev/src/components/mdx-page-content.tsx

```diff
@@ -1,16 +1,132 @@
-// @todo update from blockprotocol
+// import throttle from "lodash/throttle";
+import { useRouter } from "next/router";
 import { MDXRemote, MDXRemoteSerializeResult } from "next-mdx-remote";
-import { FunctionComponent } from "react";
+import { FunctionComponent, useEffect, useMemo, useRef, useState } from "react";
 
 import { mdxComponents } from "../util/mdx-components";
+import { Heading, PageHeadingsContext } from "./context/page-headings-context";
 
 type MdxPageContentProps = {
   serializedPage: MDXRemoteSerializeResult<Record<string, unknown>>;
 };
 
+let detectHeadingFromScrollTimer: NodeJS.Timeout | undefined = undefined;
+
 export const MdxPageContent: FunctionComponent<MdxPageContentProps> = ({
   serializedPage,
 }) => {
-  // @ts-expect-error @todo fix this
-  return <MDXRemote {...serializedPage} components={mdxComponents} />;
+  const router = useRouter();
+
+  const [headings, setHeadings] = useState<Heading[]>([]);
+
+  const detectHeadingFromScroll = useRef<boolean>(true);
```

Comment:
What is the purpose of this ref? It gets set to `false` and `true`, including by using the timer above, but the value is never read for anything.